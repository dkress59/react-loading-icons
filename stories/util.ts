import { ArgTypes } from '@storybook/react'
import { SVGProps } from 'react'

export type IconProps = JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
export const IconFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: 'transparent',
	strokeOpacity: 1,
	width: 'auto',
}
export const IconStroke: IconProps = {
	fill: 'transparent',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}
export const IconStrokeFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}

export const IconFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: 'transparent',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: 'transparent',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
