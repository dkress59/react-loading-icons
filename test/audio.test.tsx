import { render, screen } from '@testing-library/react'
import React from 'react'

import Audio from '../src/components/audio'
import { testProps } from './util'

describe('<Audio /> component', () => {
	it('renders and matches snapshot', () => {
		render(<Audio data-testid="icon" />)
		const svg = screen.getByTestId('icon')
		expect(svg).not.toBeNull()
		expect(svg).toMatchSnapshot()
		expect(svg.getAttributeNames()).toContain('viewBox')
		expect(svg.getAttribute('class')).toBe('icon-loading')
	})
	it('inherits props', () => {
		render(<Audio {...{ ...testProps, speed: undefined }} />)
		const svg = screen.getByTestId('icon')
		expect(svg).toMatchSnapshot()
		expect(svg.getAttribute('height')).toBe('2em')
		expect(svg.getAttribute('width')).toBe('auto')
		expect(svg.getAttribute('class')).toBe('icon-loading test')
	})
})
