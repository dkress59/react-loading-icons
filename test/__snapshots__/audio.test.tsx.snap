// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Audio /> component inherits props 1`] = `
<svg
  class="icon-loading test"
  data-testid="icon"
  height="2em"
  viewBox="0 0 55 80"
  width="auto"
>
  <g
    transform="matrix(1 0 0 -1 0 80)"
  >
    <rect
      fill="#000"
      fill-opacity=".5"
      height="20"
      rx="3"
      stroke="#fff"
      stroke-opacity=".5"
      width="10"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="4.3s"
        repeatCount="indefinite"
        values="20;45;57;80;64;32;66;45;64;23;66;13;64;56;34;34;2;23;76;79;20"
      />
    </rect>
    <rect
      fill="#000"
      fill-opacity=".5"
      height="80"
      rx="3"
      stroke="#fff"
      stroke-opacity=".5"
      width="10"
      x="15"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="2s"
        repeatCount="indefinite"
        values="80;55;33;5;75;23;73;33;12;14;60;80"
      />
    </rect>
    <rect
      fill="#000"
      fill-opacity=".5"
      height="50"
      rx="3"
      stroke="#fff"
      stroke-opacity=".5"
      width="10"
      x="30"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="1.4s"
        repeatCount="indefinite"
        values="50;34;78;23;56;23;34;76;80;54;21;50"
      />
    </rect>
    <rect
      fill="#000"
      fill-opacity=".5"
      height="30"
      rx="3"
      stroke="#fff"
      stroke-opacity=".5"
      width="10"
      x="45"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="2s"
        repeatCount="indefinite"
        values="30;45;13;80;56;72;45;76;34;23;67;30"
      />
    </rect>
  </g>
</svg>
`;

exports[`<Audio /> component renders and matches snapshot 1`] = `
<svg
  class="icon-loading"
  data-testid="icon"
  height="80"
  viewBox="0 0 55 80"
  width="55"
>
  <g
    transform="matrix(1 0 0 -1 0 80)"
  >
    <rect
      fill="#fff"
      height="20"
      rx="3"
      stroke="transparent"
      width="10"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="4.3s"
        repeatCount="indefinite"
        values="20;45;57;80;64;32;66;45;64;23;66;13;64;56;34;34;2;23;76;79;20"
      />
    </rect>
    <rect
      fill="#fff"
      height="80"
      rx="3"
      stroke="transparent"
      width="10"
      x="15"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="2s"
        repeatCount="indefinite"
        values="80;55;33;5;75;23;73;33;12;14;60;80"
      />
    </rect>
    <rect
      fill="#fff"
      height="50"
      rx="3"
      stroke="transparent"
      width="10"
      x="30"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="1.4s"
        repeatCount="indefinite"
        values="50;34;78;23;56;23;34;76;80;54;21;50"
      />
    </rect>
    <rect
      fill="#fff"
      height="30"
      rx="3"
      stroke="transparent"
      width="10"
      x="45"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="2s"
        repeatCount="indefinite"
        values="30;45;13;80;56;72;45;76;34;23;67;30"
      />
    </rect>
  </g>
</svg>
`;
